### DISCUSSION TITLE
Use zero-values rather than pointers

### DISCUSSION THREAD

**Discussion by alexec** [oapi-codegen/oapi-codegen#1860] [Q&A]
My generated code contains pointers when it would be preferable to have zero values:

```go
WorkspaceFlavorId *string   `json:"workspaceFlavorId,omitempty"`
```

vs

```go
WorkspaceFlavorId string   `json:"workspaceFlavorId,omitempty"`
```

How do I get the latter to work?

**Comment by jamietanna** [comment]
Via https://github.com/oapi-codegen/oapi-codegen?tab=readme-ov-file#openapi-extensions you may be looking for `x-go-type-skip-optional-pointer`? 

It's done on a per-field level which you can modify either inline or on https://github.com/oapi-codegen/oapi-codegen?tab=readme-ov-file#modifying-the-input-openapi-specification

https://github.com/oapi-codegen/oapi-codegen/pull/1694 is a way to set this globally but no ETA on merge

**Comment by alexec** [comment]
I understand that there are options for specific fields, but I have a long OpenAPI spec with many types. I don't want to have to annotate every field, for example. Can I apply this globally?

**Reply by jamietanna** [reply]
https://github.com/oapi-codegen/oapi-codegen/pull/1694 will add support for this

**Comment by jamietanna** [comment]
https://github.com/oapi-codegen/oapi-codegen/issues/1899 is another option that may come in help here
